src/main/java/org/truffleruby/options/CommandLineParser.java

```diff
@@ -443,9 +436,9 @@ private void processArgument() {
                             break;
                         }
                     }
-                    throw new MainExitException(1, "truffleruby: unknown option " + argument);
+                    throw new CommandLineException("unknown option " + argument);
                 default:
-                    throw new MainExitException(1, "truffleruby: unknown option " + argument);
+                    throw new CommandLineException("unknown option " + argument);
```

Comment:
Hmm yeah these are actual exceptions in Ruby. Can we think about that separately though. This is a fix.